### PR TITLE
[MIG] project_purchase_link: Migration to 13.0

### PR DESCRIPTION
**Pull request by tonygalmiche**

### CODE REVIEW

**Comment by tonygalmiche** [comment]
I don't know what i need to do to fix this in travis : 
`flake8 except __init__.py................................................Failed`

Thank you in advance

**Comment by emagdalenaC2i** [comment]
It is related with the pre-commit. See https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#technical-method-to-migrate-a-module-from-120-to-130-branch

- Install pre-commit once in your equipment (in Linux)
- Run pre-commit install -f in the repo.
- Run pre-commit run -a in the repo.
- Commit

**Comment by tonygalmiche** [comment]
Hi @emagdalenaC2i,
I followed this procedure : https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0
but i don't understand what i need to do to fix the problem : 
image

Thank you in advance

File: project_purchase_link/tests/test_project_purchase_link.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+from odoo.tests import common
+from odoo.tools.safe_eval import safe_eval
+
+
+class TestProjectPurchaseUtilities(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestProjectPurchaseUtilities, cls).setUpClass()
+        cls.project_model = cls.env["project.project"]
+        cls.project = cls.project_model.create({"name": "Test Project",})
```
Comment (emagdalenaC2i) [inline project_purchase_link/tests/test_project_purchase_link.py]:
```suggestion
        cls.project = cls.project_model.create({"name": "Test Project", })
```
or 
cls.project = cls.project_model.create({"name": "Test Project"})

File: project_purchase_link/tests/test_project_purchase_link.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        cls.purchase_model = cls.env["purchase.order"]
+        cls.purchase = cls.purchase_model.search(
+            [("state", "in", ("draft", "sent")),], limit=1
```
Comment (emagdalenaC2i) [inline project_purchase_link/tests/test_project_purchase_link.py]:
```suggestion
            [("state", "in", ("draft", "sent")), ], limit=1
```
or 
[("state", "in", ("draft", "sent"))], limit=1

File: project_purchase_link/tests/test_project_purchase_link.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        )
+        cls.invoice_model = cls.env["account.invoice"]
+
+    def test_project_purchase(self):
+        self.assertTrue(self.purchase)
+        self.purchase.write(
+            {"invoice_status": "to invoice",}
```
Comment (emagdalenaC2i) [inline project_purchase_link/tests/test_project_purchase_link.py]:
```suggestion
            {"invoice_status": "to invoice", }
```
or
            {"invoice_status": "to invoice"}

File: project_purchase_link/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2018 Gontzal Gomez - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+{
+    "name": "Project Purchase Link",
+    "version": "13.0.1.0.0",
+    "license": "AGPL-3",
+    "depends": ["project", "purchase", "hr_timesheet",],
```
Comment (emagdalenaC2i) [inline project_purchase_link/__manifest__.py]:
```suggestion
    "depends": ["project", "purchase", "hr_timesheet", ],
```

File: project_purchase_link/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    "author": "AvanzOSC, " "Odoo Community Association (OCA)",
+    "website": "http://github.com/OCA/project.git",
+    "category": "Project",
+    "data": ["views/project_project_view.xml",],
```
Comment (emagdalenaC2i) [inline project_purchase_link/__manifest__.py]:
```suggestion
    "data": ["views/project_project_view.xml", ],
```

**Review by emagdalenaC2i — COMMENTED** [review]
See travis, there are a few E231 missing whitespace after ','

File: project_purchase_link/models/project_project.py [hunk 6]
```diff
@@ -0,0 +1,107 @@
+# Copyright 2019 Oihane Crucelaegui - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import _, api, fields, models
```
Comment (emagdalenaC2i) [inline project_purchase_link/models/project_project.py]:
```suggestion
from odoo import _, fields, models
```

Comment (tonygalmiche) [reply project_purchase_link/models/project_project.py]: Thank you very much @emagdalenaC2i  for all of these answers but there is still a problem and i don't see which one.

when i run flake8 on my computer i get errors but i don't know what to do : 
```
$ flake8 project_purchase_link
project_purchase_link/__init__.py:4:1: F401 '.models' imported but unused
project_purchase_link/tests/__init__.py:4:1: F401 '.test_project_purchase_link' imported but unused
project_purchase_link/models/__init__.py:4:1: F401 '.project_project' imported but unused
```

Comment (tonygalmiche) [reply project_purchase_link/models/project_project.py]: now the pre-commit works on my computer : 
```
13:25 $ pre-commit run -a
black....................................................................Passed
prettier.................................................................Passed
prettier xml plugin......................................................Passed
eslint...............................................(no files to check)Skipped
- hook id: eslint
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Debug Statements (Python)................................................Passed
Fix python encoding pragma...............................................Passed
Check for case conflicts.................................................Passed
Check docstring is first.................................................Passed
Check that executables have shebangs.................(no files to check)Skipped
Check for merge conflicts................................................Passed
Check for broken symlinks............................(no files to check)Skipped
Check Xml................................................................Passed
Mixed line ending........................................................Passed
flake8 except __init__.py................................................Passed
flake8 only __init__.py..................................................Passed
pylint with optional checks..............................................Passed
- hook id: pylint
- duration: 2.29s
pylint with mandatory checks.............................................Passed
pyupgrade................................................................Passed
isort except __init__.py.................................................Passed
Generate default setup.py for an addons directory........................Passed
✔ ~/docker-compose/OCA/project_purchase_link-13/addons/project [13.0-mig-project_purchase_link|✔] 
```

**Comment by tonygalmiche** [comment]
I disabled unit tests because I don't know how to make them work. I want help on this subject
I think it's ready for reviews
Thank you in advance

**Review by HaraldPanten — APPROVED** [review]
Functional review 👍 

Just a comment, not blocking: IMO, you could improve the README as it is too short. New users should be informed that they have to activate analytic accounting and assign the correct analytic account to link projects and purchases.

**Comment by tonygalmiche** [comment]
Thank you @HaraldPanten for review
I added USAGE.rst

**Comment by HaraldPanten** [comment]
Now looks perfect. Thanks!

**Comment by emagdalenaC2i** [comment]
> I disabled unit tests because I don't know how to make them work. I want help on this subject

You should keep the tests. If you don't know how it works it is a great opportunity to learn something new.
You can try to uncomment some lines and see how it works and the specific warnings and errors with each line

**Comment by tonygalmiche** [comment]
Hi @emagdalenaC2i,
That's what I did, but since I'm not used to unit testing and the billing module has evolved a lot, I had to redo the tests differently. But now there are as many tests as before.

I also tried to make a git rebase, but since I'm not used to it either, I'm not sure if it's correct.
Thanks in advance for the verification

**Comment by tonygalmiche** [comment]
I just did a git rebase again because it didn't work. It should be good now

**Review by emagdalenaC2i — APPROVED** [review]
LGTM

**Comment by tonygalmiche** [comment]
Thank you @emagdalenaC2i  for your review. It was my first migration :)

**Comment by pedrobaeza** [comment]
@dreispt ocabot stuck on merge

**Comment by pedrobaeza** [comment]
@sbidoul another one that ocabot doesn't finish the merge. Can you please check?

**Comment by sbidoul** [comment]
@pedrobaeza Travis did not report back on the last two commits of the merge branch. 
I'm hearing more and more issues with travis-ci.org. So I migrated OCA/project* to travis-ci.com.
Let's see

/ocabot merge nobump

**Pull request merged**